Code for the Old God! Old Cult is back!

Robustin opened the pull request:
Old Cult has returned to its last form prior to removal,  with just a few tweaks:
- The "X acolytes must escape" objective has been disabled.
- Invoking the imbue talisman will consume the imbue talisman (RIP talisman factories)
- Invoking a stun talisman will brute the caster for 15, up from 10
- Stun talisman's stun duration reduced to 9 tics, down from 10
- Stun talisman no longer mutes for the full duration, now applies 4 tics of mute, down from 10, and 15 tics of stutter and a _new_ cultslur proc that should make it difficult to decipher radio messages for help, but with the downside of making it obvious that the victim isn't merely drunk
- Cult retains the timestop rune, requiring 3 casters (fun rune, can't be as OP as blood boil anyway)
- Experimental: Drawing a Nar-Sie rune will send out a notification similar to newcult's. Drawing the nar-sie rune now takes 10x the time of a normal rune. If coded properly (hah) it should name the cultist who is drawing the rune and their location, giving the crew a small window to intervene before the ritual is complete (invoking the rune is still the same and will occur instantly).

Robustin committed:
Update mind.dm

Robustin committed:
Update cult.dm

Robustin committed:
Update cult_items.dm

Robustin committed:
Update cult_structures.dm

Robustin committed:
Update ritual.dm

Robustin committed:
Update runes.dm

Robustin committed:
Update talisman.dm

Robustin committed:
Update soulstone.dm

Robustin committed:
Update one_click_antag.dm

Robustin committed:
Update species.dm

Robustin committed:
Update mob_defines.dm

Robustin committed:
Update paper.dm

Robustin committed:
Update narsie.dm

Robustin committed:
Update other_reagents.dm

Robustin committed:
Create Robustin-Oldcult

Robustin committed:
Update mob_helpers.dm

Robustin committed:
Update talisman.dm

Robustin committed:
Update mob_helpers.dm

Robustin committed:
Update talisman.dm

Robustin committed:
Update Robustin-Oldcult

Robustin committed:
Update runes.dm

Robustin committed:
Update ritual.dm

Robustin committed:
Update ritual.dm

KorPhaeron commented on code/game/gamemodes/cult/cult.dm:
```diff
@@ -59,8 +82,8 @@
 	//cult scaling goes here
-	recommended_enemies = 1 + round(num_players()/8)
-	orbs_needed = recommended_enemies
+	recommended_enemies = 3 + round(num_players()/20)
```
>scaling starting cultists when their entire objective is to have X people and the round ends

At least it's a bit more sane than the previous oldcult scaling that would routinely give you 8-9 people I guess

Robustin replied:
The "survive" objective is gone.

KorPhaeron replied:
Yeah but the nar-sie stand in a square with 9 guys isn't

KazeEspada replied:
We make a bigger nar nar rune and make it require more people.

KazeEspada commented:
Dont remove new cult. Do both as seperate gamemodes.

KorPhaeron commented on code/modules/power/singularity/narsie.dm:
```diff
@@ -61,7 +61,7 @@
 /obj/singularity/narsie/proc/godsmack(atom/A)
-	if(isobj(A))
+	if(istype(A,/obj/))
 		var/obj/O = A
```
Why'd you change this back?

duncathan replied:
He probably doesn't know what any of it does

Robustin replied:
Ding ding ding @duncathan gets it.

I'll change this back if doesn't screw with oldcult.

KazeEspada commented:
Also summon nar nar should always be an objective. And nar nar consuming your target is a sac.

KorPhaeron commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -183,10 +183,13 @@
 		M.confused += 3
-		if((is_handofgod_cultist(M) && !is_handofgod_prophet(M)))
+		if(iscultist(M) || (is_handofgod_cultist(M) && !is_handofgod_prophet(M)))
+			ticker.mode.remove_cultist(M.mind)
```
I think security mass arrest+deconvert is dull gameplay personally.

If we gotta have conversion back I'd at least prefer no deconversion.

Robustin replied:
The deconversion takes more effort than almost any other. You have to babysit the cultists and keep them separated since a single loose cultist with a tome can undo everything. You also also have to babysit if you want to confirm that they've been deconverted since theres only a brief moment when you can tell that a cultist is purified. 

Also being able to summon cultists during this process makes it super risky as opposed to shooting them on the spot.

KorPhaeron commented on html/changelogs/Robustin-Oldcult:
```diff
@@ -0,0 +1,39 @@
+  - tweak: "Invoking the stun talisman now hurts the cultist more"
+  - tweak: "The stun talisman duration is nerfed by 10%, mute is nerfed by 60% but a stutter and special new slur effect will last much longer"
+  - tweak: "The old cult has retained the rune of time stop, now requiring 3 cultists to invoke"
+  - rscdel: "New Cult is removed!"
```
In the future you can do changelogs from the PR body

optimumtact replied:
nothing wrong with this way though

Robustin committed:
Update narsie.dm

Robustin commented:
The dreaded build failure. Still have no idea how to narrow down what the problem is. Changed a lot of lines and I'm sure there's some human error in there.

KazeEspada commented:
code/game/gamemodes/cult/cult.dm:31:error: bad argument definition
code/game/gamemodes/cult/cult.dm:33:error: bad argument definition
code/game/gamemodes/cult/cult.dm:43:error: inconsistent indentation
code/game/gamemodes/cult/cult.dm:45:error: inconsistent indentation
code/game/gamemodes/cult/cult.dm:44:warning: empty 'else' clause
code/game/gamemodes/cult/cult.dm:50:error: bad argument definition
code/game/gamemodes/cult/ritual.dm:142:error: .: expected end of statement
code/game/gamemodes/cult/ritual.dm:142:error: missing left-hand argument to <.

Robustin committed:
Update ritual.dm

Robustin committed:
Update ritual.dm

Robustin committed:
Update cult.dm

Razharas commented:
@Robustin 
Maybe you should try to finish your carp karate pr first?
THis level of coding seem above your current coding powerlevel

KazeEspada commented:
Gamemode code is a monstrosity(which is why no one reverted cult already).

ChangelingRain commented:
Get in coderbus so we can yell at you okay

Robustin committed:
Update cult.dm

Robustin committed:
Update runes.dm

Buggy123 commented:
:+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1::+1:
>50% of the server has been begging for this, thank you.

Robustin committed:
Update runes.dm

optimumtact commented:
Why was this closed?

xxalpha commented:
Because memetainers.

Robustin committed:
Update runes.dm

PKPenguin321 commented:
Not the solution to stun talismans that I would use personally, but :+1:

ChangelingRain commented:
Cheridan said I could.
That, and I want robustin to get into coderbus.

Shadowlight213 commented on code/game/gamemodes/cult/talisman.dm:
```diff
@@ -208,6 +244,39 @@ Rite of Disorientation
+		
+/obj/item/weapon/paper/talisman/stun/attack(mob/living/target, mob/living/user)
+	if(iscultist(user))
+-		user.whisper(invocation)
```
More extra minuses.

Shadowlight213 commented on code/modules/reagents/chemistry/reagents/other_reagents.dm:
```diff
@@ -183,10 +183,13 @@
 		M.stuttering += 4
 		M.Dizzy(5)
+		if(iscultist(M) && prob(5))
+-			M.say(pick("Av'te Nar'sie","Pa'lid Mors","INO INO ORA ANA","SAT ANA!","Daim'niodeis Arc'iai Le'eones","R'ge Na'sie","Diabo us Vo'iscum","Eld' Mon Nobis"))
```
Remove extra minus

Shadowlight213 commented on code/game/gamemodes/wizard/soulstone.dm:
```diff
@@ -123,6 +123,12 @@
 		if("VICTIM")
 			var/mob/living/carbon/human/T = target
+			if(ticker.mode.name == "cult" && T.mind == ticker.mode:sacrifice_target)
+-				if(iscultist(user))
```
Delete extra minuses

Robustin committed:
Update talisman.dm

Robustin committed:
Update other_reagents.dm

Robustin committed:
Update soulstone.dm

Robustin commented:
Made all the fixes to the code, still waiting for @ChangelingRain to re-open. Literally dragged onto IRC so Cheridan and Joan can whine about stun talismans and Nar-Sie as an objective when this PR addresses both in a way the should improve oldcult.

ChangelingRain commented:
Robustin has sat around in coderbus for a while.

Robustin committed:
Update human.dm

Robustin commented:
Why in the world am I getting Travis Errors on files I haven't even touched. Inconsistent Indentation on Human.dm?

Plz

Shadowlight213 commented on code/modules/mob/living/carbon/human/human.dm:
```diff
@@ -706,7 +706,7 @@
 /mob/living/carbon/human/singularity_pull(S, current_size)
 	if(current_size >= STAGE_THREE)
-		var/list/handlist = list(l_hand, r_hand)
+	var/list/handlist = list(l_hand, r_hand)
```
You accidentally unidentified this. That's the source of your error

Robustin replied:
I intentionally did it to see if I can't get rid of whatever first triggered that error.

Robustin committed:
Update human.dm

duncathan commented:
Cold for the Old Gold! Old Cold is Bold!

Zilenan91 commented:
The main reason oldcult was removed in the past was because stunpaper was ultimately too effective. I still say the best place for it would be in supply talismans only to carry cultists through the early game and nothing more.

duncathan commented:
Actually zilenan the main reason old cult was removed was because your incessant whining actually convinced us that people hated it

Zilenan91 commented:
More than a little of that yeah.

drynwyn commented:
holy shit did zile"Whining about something constantly until it's changed is the only way to do anything here"nan just acknowledged fault

Zilenan91 commented:
I've literally never once made it a secret that I've done these things.

Core0verload commented:
Fuck yeah. :+1: 

The only thing i'm concerned about is communication runes. Do they still have multiple uses or do they disappear after the first one?

Robustin commented:
Communication rune functionality was moved to the tomes a while ago. You can communicate using your tome or (if you lose it) you can damage yourself to send out a "flawed" message. 

Still not sure how using the web editor has pissed off a random line of some random code I didn't touch but I'm poking around for answers.

Robustin committed:
Update human.dm

Steelpoint commented:
If Oldman can get this to work then I support it and I may turn Cultist back on as a antag option.

Also I see no problem with allowing Sec to deconvert a cultist, as he noted its a massive timesink and has a high resource requirment that needs you to babysit that one person for several minutes, and its risky to the point you're better off executing them, but the reward is a new loyal crewmember.

Robustin committed:
Update species.dm

Robustin committed:
Update mind.dm

ShadowDeath6 commented:
The inglorious return of stun talismans
And the glorious return of everything else
I missed oldcult :+1:

Robustin committed:
Update cult.dm

Robustin committed:
Update ritual.dm

Shadowlight213 commented on code/modules/mob/mob_defines.dm:
```diff
@@ -79,6 +79,7 @@
 	var/obj/item/r_hand = null//Living
 	var/obj/item/weapon/storage/s_active = null//Carbon
 
+	var/seer = 0 //for cult//Carbon, probably Human
```
Where is this actually used? Just use see_ov

Robustin replied:
One of the many lines where I said "why does this exist?" and answered "I dont know, better add it anyway".

Robustin committed:
Update talisman.dm

Robustin committed:
Update runes.dm

Robustin committed:
Update mob_defines.dm

Shadowlight213 commented:
@Robustin I downloaded your pr and am working on fixing all the errors. Gimme a bit and I'll PR it to your branch.

Shadowlight213 committed:
Updates a lot of stuff making it compile hopefully

Shadowlight213 committed:
Fixes Merge conflicts

Robustin committed:
Update cult.dm

Robustin commented:
Hah, I'd appreciate it. I haven't touched code in ~10 years but I never remember C++/Java being this bitchy about indentation. 

I think I got most of the cult.dm stuff.

Shadowlight213 committed:
more merge conflict fixing

RemieRichards commented:
@Robustin 
Most C family languages use {} for deciding where a block of code starts or ends.
Other Languages use Whitespace, the level of indentation to decide where a block starts or end.
While you can indent in C languages (and its strongly encouraged) its purely for readability there.

Byond is sort of a C family language, and does in fact support {}, but we don't use that style, so you have to get used to proper indentation.

Shadowlight213 commented:
@Robustin Ok. I made a pr to your branch. Should compile now.

Robustin committed:
Update species.dm

Robustin committed:
Update talisman.dm

Shadowlight213 commented:
@Robustin STOP MAKING MORE CHANGES, REQUIRING ME TO MAKE EVER MORE MERGE CONFLICT FIXES!!! REEEEEEEEEEEEEE

Robustin commented:
wew, ok ok.

Shadowlight213 committed:
even more merge conflict fixes

Shadowlight213 committed:
Jitter fix

Shadowlight213 commented:
@Robustin Ok. everything fixed and it compiles locally for me.

Robustin committed:
Merge pull request #3 from Shadowlight213/oldcultrobustin

Makes old cult compile

Robustin commented:
Woo green check mark. Thanks to @Shadowlight213 for hammering out the compiling issues. Not even the cult can wield whatever dark magicks are necessary to appease Travis.

Bawhoppen commented:
:+1:x:100: praise our savior

Robustin commented:
>40 person newcult round
>NarSie summoning begun 10 minutes into the round
>Anyone who comes near gets the ol' stun and instantly sacrificed at full HP meme, instantly becomes artificer to rinse and repeat this meme

Plz merge.

Iamgoofball commented:
@Incoming5643

Incoming5643 commented:
No problems here from me, next time I do real work on datum antags (soon™.) it'll have to start with vigorous conflict resolution anyways, and I was going to throw out the little work I had done on cult stuff anyway

Shadowlight213 commented:
@Robustin you'll need to ask someone for a test merge on the server if you haven't already. While I did get the code to compile, I didn't have time to actually run it locally.
I would not recommend it being actually merged until it's tested.

PKPenguin321 commented:
merge conflicts reeee

Shadowlight213 commented:
I'm going to be expected to fix these conflicts, aren't I?
Sigh

Shadowlight213 committed:
Merge conflicts fix

Shadowlight213 commented:
Ok, I fixed the merge conflicts, but git freaked out with the commit log on pulling from master.
https://github.com/Robustin/-tg-station/pull/4

Robustin committed:
Merge pull request #4 from Shadowlight213/oldcultrobustin1

Fixes more merge conflicts.

Shadowlight213 commented:
Travis actually technically passed. It failed on dreamstation for some reason.

duncathan commented:
Gotcha

Shadowlight213 commented:
So. test merge on sybil. A few issues.
Cultists aren't assigned correctly at round start. while traitor panel shows them as cultists, they dont have any of the cultist gear, nor their antag hud. deculting and reculting them via traitor panel fixes that.
No end round report.
The cultist amulet paper doesn't have a spawn tome function. it seems to be for newcult.

phil235 commented:
>55 commits

Please squash your commits...

Robustin commented:
Ugh, the amulet stuff was def. in the oldcult code I copied most of this from. I don't even have much experience myself with it. That's not a serious concern though as long as cultists get their tomes, another cultist can always summon a new tome if you lose yours. 

Fixing the assignment stuff is going to be a lot harder.

Robustin committed:
Should fix item spawning

Uses the code newcult was using to give tome/talisman, but no sacrificial dagger.

Also raises minimum crew size for cult to 35 (its not a challenge when your sac target is also the only sec member), and decreases scaling so that the range is 4 cultists, 5 when over 60pop, 6 when over 90 pop.

Shadowlight213 commented:
I'm fairly certain the HUDs not showing up means that they aren't getting their antag status properly assigned or something. From what I observed I don't think the cultists got their intro text either as they didn't seem to realize they were cult until I reapplied the status. However, they still were on the check antag list, as well as having cultist prayers and having the cultist antag flag in the traitor panel.
I'm guessing a setup cultist proc is failing to be called somewhere for the roundstart cultists beyond setting their special role to cultist and adding them to the cultist list.

Shadowlight213 commented on code/game/gamemodes/cult/cult.dm:
```diff
@@ -75,18 +99,40 @@
+
 /datum/game_mode/cult/post_setup()
 	modePlayer += cult
-	for(var/datum/mind/cult_mind in cult)
```
Why did you remove this code block? This is what gave the original cultists their hud, objectives, gear, etc.
Move the objective generation code above this.

Robustin replied:
There were about 1,000 lines to compare in addition to what I added, I guess it was an inadvertent cut. I'd go and check for sure but I'll just take your word for it.

Shadowlight213 replied:
I already checked before I posted. :P

Robustin committed:
Fixes cultists not getting objectives/HUD/etc.

Robustin commented:
Ran it locally, everything looked in order. Hud, items, etc. was there like normal and im happy to report the narsie rune works exactly as i had hoped.

Core0verload commented:
Mergetest when?

Shadowlight213 committed:
Adds delay for cultists removing Narsie rune.
Adds logging to removing a narsie rune
Fixes cultslurring not decreasing
Fixes supply talisman not giving tomes and removes tome from cult starting gear.

Shadowlight213 commented:
Added some logging, as well as a delay for removing the narsie rune for cultists.
Fixed tome spawning from supply amulets and cultslur decreasing.
https://github.com/Robustin/-tg-station/pull/5

Robustin committed:
Merge pull request #5 from Shadowlight213/oldcultrobustin2

More cult fixes.

Robustin commented:
Squashing commits has proven to be an intractable nightmare.

Merge and end this.

KorPhaeron commented:
No need for two of the same PR
